src/plan/global.rs

```diff
@@ -441,6 +442,9 @@ impl<VM: VMBinding> BasePlan<VM> {
         global_side_metadata_specs: Vec<SideMetadataSpec>,
     ) -> BasePlan<VM> {
         let stats = Stats::new(&options);
+        use crate::util::constants::DEFAULT_STRESS_FACTOR;
+        let is_stress_test_gc_enabled = *options.stress_factor != DEFAULT_STRESS_FACTOR
```

Comment:
I am guessing you were trying to optimize this. But this happens when we create `MMTK`, and options can be set after creating `MMTK`. If options are set after creating `MMTK`, this value could be incorrect.

There are a few things you can do to fix this:
1. revert the change, and always check the option where it is used. 
2. do not allow those options to be set by command line arguments (we do have a few options that cannot be set through command line args - but this is not preferred, and will be fixed at some point). 
https://github.com/mmtk/mmtk-core/blob/3dbdd7ae1076fb022f483b8b91cf37425dc6f0ee/src/util/options.rs#L341-L344
3. add another option, something like `enable_stress_gc`. The two factor options will only be used when `enable_stress_gc` is true.

Reply:
Ah sorry I didn't realize that would be an issue. I noticed that the check is potentially expensive since it is three loads + 1 comparison. Its performance does not matter in `alloc_slow_inline()`, which is the only other place it is used. However, for the intermediate path, its performance overhead was noticeable. Hence, I optimized it to be 1 load instead.

Hmm... I haven't looked into the option setting code yet, but would it not be easier/better to just update the new boolean field if and when the commandline stress factor option is processed?

Reply:
> would it not be easier/better to just update the new boolean field if and when the commandline stress factor option is processed?

I feel that would be worse. That would mean the options need to know that it is being cached in the plan (so it can update the value in the plan), and it introduces unnecessary dependency. You could just add a new option `enable_stress_gc`.

Reply:
Right I see. In that case, what is the semantics of `enable_stress_gc`? If `enable_stress_gc=false` (I guess which is the default?) but `stress_factor=8192` then will we stress test or not?

Reply:
I assume no. If we do not enable stress gc, we do not further check the factors (it would be good to give a warning if stress/analysis factors are set but stress gc is disabled).

Reply:
All what I am saying above is under the assumption that the old code of checking actually has an overhead. If it does not have overhead, we could just check against the factor options. You could simply revert this change first, and run one or two benchmarks against the build here: https://github.com/mmtk/mmtk-core/pull/571#issuecomment-1104629725, and see if there is any overhead at all.

Reply:
Alright. Did you mean this build (i.e. the one with `unlikely()`): https://github.com/mmtk/mmtk-core/pull/571#issuecomment-1120199298?

Reply:
Umm, using the later build with `unlikely` seems better. You just need to figure out the cost of 'cached boolean' vs 'accessing option directly', with or without `unlikely()` (but it would be better with unlikely, as you said). You could just pick one or two benchmarks that seemed to have a slowdown in https://github.com/mmtk/mmtk-core/pull/571#issuecomment-1104629725.

Reply:
https://squirrel.anu.edu.au/plotty/kunals/bench-test/#0|2022-06-03-immix-stress-fix-shrew-2022-06-03-Fri-155155&benchmark^build^invocation^iteration&time^time.other^time.stw&|10&iteration^1^4&benchmark^2^cassandra&benchmark^2^h2|20&1^invocation|70&build^immix-stress-pr|30&1&benchmark&build;jdk11-master-is_stress-core_28e1af1d-jdk_040da09-vm_ca90b43f0f5|41&Histogram%20(with%20CI)^build^benchmark&

Both builds have minimal, if any, impact on the mutator time. I'll stick with the no cached boolean that is currently on master then.